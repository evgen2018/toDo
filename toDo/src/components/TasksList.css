.droppable-block{
    width: 600px;
    margin: 0 auto;
}