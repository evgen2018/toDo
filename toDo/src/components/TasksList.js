import React from 'react';
import PropTypes from 'prop-types';
import TaskItem from "./TaskItem";
import omit from 'object.omit';
import './TasksList.css';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
import reorder from '../helpers/reorder';
import {getItemStyle, getListStyle} from '../helpers/changeDrugStyle';

class TaskList extends React.Component{
    static propTypes = {
        tasks: PropTypes.array.isRequired,
        selected: PropTypes.array.isRequired,
        toggleSelectTask: PropTypes.func.isRequired,
        deleteTask: PropTypes.func.isRequired,
        editTask: PropTypes.func.isRequired,
        reOrderTasks: PropTypes.func.isRequired
    };

    constructor(props) {
        super(props);
        this.state = {
            tasks: props.tasks,
            isAnyTaskEditing: false
        };
        this.onDragEnd = this.onDragEnd.bind(this);
    }

    onDragEnd(result) {
        // dropped outside the list
        if (!result.destination) {
            return;
        }

        const tasks = reorder(
            this.state.tasks,
            result.source.index,
            result.destination.index
        );

        this.props.reOrderTasks(tasks);

        this.setState({
            tasks
        });
    }

    toggleEditing = () =>{
        this.setState({isAnyTaskEditing:!this.state.isAnyTaskEditing})
    };

    componentWillReceiveProps(nextProps) {
        if(this.props !==nextProps) {
            this.setState({
                tasks: nextProps.tasks
            });
        }
    }

    render(){
        const props = omit(this.props, 'tasks');

        return (
                <DragDropContext onDragEnd={this.onDragEnd}>
                    {this.state.tasks.length > 0 &&
                        <Droppable droppableId="droppable" isDropDisabled={this.state.isAnyTaskEditing}>
                            {(provided, snapshot) => (
                            <div
                                ref={provided.innerRef}
                                className='droppable-block'
                                style={getListStyle(snapshot.isDraggingOver)}>
                                {this.state.tasks.map((taskData, index) => (
                                    <Draggable key={taskData.id} draggableId={taskData.id} index={index}>
                                        {(provided, snapshot) => (
                                            <div
                                                ref={provided.innerRef}
                                                {...provided.draggableProps}
                                                {...provided.dragHandleProps}
                                                style={getItemStyle(
                                                    snapshot.isDragging,
                                                    provided.draggableProps.style
                                                )}>
                                                <TaskItem
                                                    taskData={taskData}
                                                    key={index}
                                                    toggleEditing={this.toggleEditing}
                                                    isAnyTaskEditing={this.state.isAnyTaskEditing}
                                                    {...props}/>
                                            </div>
                                        )}
                                    </Draggable>
                                ))}
                                {provided.placeholder}
                            </div>
                            )}
                        </Droppable>
                    }
                </DragDropContext>
            )
    }
}

export default TaskList
