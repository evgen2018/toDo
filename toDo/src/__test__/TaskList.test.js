import React from 'react';
import TasksList from '../components/TasksList';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('TaskList list test', () => {
    const props = {
        tasks: [{id:1, title:'mockTitle'}, {id:2, title:'mockTitle'}],
        selected:[],
        toggleSelectTask: ()=>{},
        deleteTask: ()=>{},
        editTask: ()=>{},
        reOrderTasks: ()=>{}
    };
    const wrapper = shallow(
        <TasksList {...props}/>
    );

    it('TaskList renders properly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });

});

