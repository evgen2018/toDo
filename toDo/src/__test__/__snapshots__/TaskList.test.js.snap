// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList list test TaskList renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskList
    deleteTask={[Function]}
    editTask={[Function]}
    reOrderTasks={[Function]}
    selected={Array []}
    tasks={
      Array [
        Object {
          "id": 1,
          "title": "mockTitle",
        },
        Object {
          "id": 2,
          "title": "mockTitle",
        },
      ]
    }
    toggleSelectTask={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(Droppable)
        direction="vertical"
        droppableId="droppable"
        ignoreContainerClipping={false}
        isCombineEnabled={false}
        isDropDisabled={false}
        type="DEFAULT"
      >
        [Function]
      </Connect(Droppable)>,
      "onDragEnd": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "direction": "vertical",
        "droppableId": "droppable",
        "ignoreContainerClipping": false,
        "isCombineEnabled": false,
        "isDropDisabled": false,
        "type": "DEFAULT",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(Droppable)
          direction="vertical"
          droppableId="droppable"
          ignoreContainerClipping={false}
          isCombineEnabled={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
        "onDragEnd": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "vertical",
          "droppableId": "droppable",
          "ignoreContainerClipping": false,
          "isCombineEnabled": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
